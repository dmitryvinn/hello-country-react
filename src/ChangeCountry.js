import React from "react";

const buttonStyle = {
  border: 'none',
  color: '#ffffff',
  padding: '15px 32px',
  textAlign: 'center',
  textDecoration: 'none',
  display: 'inlineBlock',
  fontSize: '16px',
  backgroundColor: '#555555'
}

class CountryFlag extends React.Component {
  render() {
    return <button style={buttonStyle} onClick={this.props.action}>Switch Country</button>
  }
}

export default CountryFlag;
