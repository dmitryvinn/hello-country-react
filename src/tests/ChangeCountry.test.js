import React from 'react';
import renderer from 'react-test-renderer';
import ChangeCountry from "../ChangeCountry";

test('Country Change Button ', () => {
    // Initialize Component
    const component = renderer.create(
        <ChangeCountry/>
    );
    // Format result
    let tree = component.toJSON();
    // Assert
    expect(tree).toMatchSnapshot();
});
