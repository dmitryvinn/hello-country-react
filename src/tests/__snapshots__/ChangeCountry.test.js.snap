// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Change Button  1`] = `
<button
  className="changeCountry"
  onClick={undefined}
  style={
    Object {
      "backgroundColor": "#555555",
      "border": "none",
      "color": "#ffffff",
      "display": "inlineBlock",
      "fontSize": "16px",
      "padding": "15px 32px",
      "textAlign": "center",
      "textDecoration": "none",
    }
  }
>
  Switch Country
</button>
`;
